source/common/formatter/substitution_formatter.cc

```diff
@@ -1541,6 +1551,36 @@ ProtobufWkt::Value PlainStringFormatter::formatValue(const Http::RequestHeaderMa
   return str_;
 }
 
+PlainNumberFormatter::PlainNumberFormatter(const double& num) { num_.set_number_value(num); }
+
+absl::optional<std::string> PlainNumberFormatter::format(const Http::RequestHeaderMap&,
+                                                         const Http::ResponseHeaderMap&,
+                                                         const Http::ResponseTrailerMap&,
+                                                         const StreamInfo::StreamInfo&,
+                                                         absl::string_view) const {
+  std::string str = std::to_string(num_.number_value());
+  str = str.substr(0, str.find_last_not_of('0') + 1);
+  str = str.substr(0, str.find_last_not_of('.') + 1);
+  return str;
```

Comment:
I think we can do better than this. Please have a look at `absl::StrFormat`.

Reply:
Done.